subsys/net/lib/download_client/src/download_client.c

```diff
@@ -457,7 +461,14 @@ void download_thread(void *client, void *a, void *b)
 				/* Restart and suspend */
 				break;
 			}
-			reconnect(dl);
+
+			err = reconnect(dl);
+			if (err == -ENETUNREACH) {
+				LOG_WRN("Network is down, "
```

Comment:
Shall we do this? I guess `ENETUNREACH` is just a temporary condition? Maybe we should just put it into sleep for some time and try again later? @lemrey

Reply:
In the case of ENETUNREACH the download client will spinnloop. I think it is better that the application takes care of this than trying to add some smart logic here. We don't know how long this intermediate condition is.

Reply:
In that case we should notify the application with `error_evt_send(dl, ENETUNREACH);` so it's aware something fishy is going on. How will the application know when to resume?

Reply:
You forgot to add `error_evt_send(dl, ENETUNREACH);` here as well.